app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt

```diff
@@ -401,28 +401,36 @@ class GleanMetricsService(private val context: Context) : MetricsService {
             Glean.registerPings(Pings)
             Glean.initialize(context, Configuration(channel = BuildConfig.BUILD_TYPE))
 
-            Metrics.apply {
-                defaultBrowser.set(Browsers.all(context).isDefaultBrowser)
-                MozillaProductDetector.getMozillaBrowserDefault(context)?.also {
-                    defaultMozBrowser.set(it)
-                }
-                mozillaProducts.set(MozillaProductDetector.getInstalledMozillaProducts(context))
-            }
-
-            SearchDefaultEngine.apply {
-                val defaultEngine = context
-                    .components
-                    .search
-                    .searchEngineManager
-                    .defaultSearchEngine ?: return@apply
+            setStartupMetrics()
+        }
+    }
 
-                code.set(defaultEngine.identifier)
-                name.set(defaultEngine.name)
-                submissionUrl.set(defaultEngine.buildSearchUrl(""))
+    /**
+     * Sets some basic Glean metrics required by Fenix.
+     * This is a separate function to simplify testing.
```

Comment:
nit: I don't think this comment is necessary. Having a helper function for this to make it more testable seems perfectly reasonable without justification :)

Reply:
Ah, good point :)